src/osd/PeeringState.cc

```diff
@@ -1855,7 +1855,7 @@ class bucket_candidates_t {
   osd_id_t pop_osd() {
     ceph_assert(!is_empty());
     auto ret = osds.back();
```

Comment:
Seems like this also needs to change?

Reply:
Whoops, I found this bug before Christmas and when I picked it back up forgot how much I'd done. Thanks!